pkg/iphelpers/iphelpers_test.go

```diff
@@ -571,6 +572,190 @@ var _ = Describe("DecIPAddress operations", func() {
+
+		_, ipnet, err := net.ParseCIDR("192.168.2.200/27")
+		ip, _ := AddressRange(ipnet)
+
+		Expect(err).NotTo(HaveOccurred())
+
+		firstip, lastip, err := GetIPRange(net.ParseIP(ip.String()), *ipnet)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(fmt.Sprint(firstip)).To(Equal("192.168.2.193"))
+		Expect(fmt.Sprint(lastip)).To(Equal("192.168.2.222"))
+
+	})
+	It("creates an IPv4 range properly for 24 bits network address", func() {
+
+		_, ipnet, err := net.ParseCIDR("192.168.2.200/24")
+		ip, _ := AddressRange(ipnet)
+
+		Expect(err).NotTo(HaveOccurred())
+
+		firstip, lastip, err := GetIPRange(net.ParseIP(ip.String()), *ipnet)
+
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(fmt.Sprint(firstip)).To(Equal("192.168.2.1"))
+		Expect(fmt.Sprint(lastip)).To(Equal("192.168.2.254"))
+
+	})
+	// Handy IPv6 CIDR calculator: https://www.ultratools.com/tools/ipv6CIDRToRangeResult?ipAddress=2001%3A%3A0%2F28
+	It("creates an IPv6 range properly for 116 bits network address", func() {
+
+		_, ipnet, err := net.ParseCIDR("2001::0/116")
+		ip, _ := AddressRange(ipnet)
+
+		Expect(err).NotTo(HaveOccurred())
+
+		firstip, lastip, err := GetIPRange(net.ParseIP(ip.String()), *ipnet)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(fmt.Sprint(firstip)).To(Equal("2001::1"))
+		Expect(fmt.Sprint(lastip)).To(Equal("2001::fff"))
+
+	})
+
+	It("creates an IPv6 range when the first hextet has leading zeroes", func() {
+
+		_, ipnet, err := net.ParseCIDR("fd:db8:abcd:0012::0/96")
+		ip, _ := AddressRange(ipnet)
+
+		Expect(err).NotTo(HaveOccurred())
+
+		firstip, lastip, err := GetIPRange(net.ParseIP(ip.String()), *ipnet)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(fmt.Sprint(firstip)).To(Equal("fd:db8:abcd:12::1"))
+		Expect(fmt.Sprint(lastip)).To(Equal("fd:db8:abcd:12::ffff:ffff"))
+
+	})
+
+	It("creates an IPv6 range properly for 96 bits network address", func() {
+
+		_, ipnet, err := net.ParseCIDR("2001:db8:abcd:0012::0/96")
+		ip, _ := AddressRange(ipnet)
+
+		Expect(err).NotTo(HaveOccurred())
+
+		firstip, lastip, err := GetIPRange(net.ParseIP(ip.String()), *ipnet)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(fmt.Sprint(firstip)).To(Equal("2001:db8:abcd:12::1"))
+		Expect(fmt.Sprint(lastip)).To(Equal("2001:db8:abcd:12::ffff:ffff"))
+
+	})
+	It("creates an IPv6 range properly for 64 bits network address", func() {
+
+		_, ipnet, err := net.ParseCIDR("2001:db8:abcd:0012::0/64")
+		ip, _ := AddressRange(ipnet)
+
+		Expect(err).NotTo(HaveOccurred())
+
+		firstip, lastip, err := GetIPRange(net.ParseIP(ip.String()), *ipnet)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(fmt.Sprint(firstip)).To(Equal("2001:db8:abcd:12::1"))
+		Expect(fmt.Sprint(lastip)).To(Equal("2001:db8:abcd:12:ffff:ffff:ffff:ffff"))
+
+	})
+	It("do not fail when the mask meets minimum required", func() {
+
+		_, badipnet, err := net.ParseCIDR("192.168.21.100/30")
+		badip, _ := AddressRange(badipnet)
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
please move this below the `net.ParseCIDR` call, from which this variable comes from.

Reply:
same